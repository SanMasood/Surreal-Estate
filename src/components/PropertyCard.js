import React from 'react';
import PropTypes from 'prop-types';

import '../styles/PropertyCard.css';

const PropertyCard = (props) => (
  <div className="property-card">
    <div className="title">
      Title:
      {props.title}
    </div>
    <div className="type">
      City:
      {props.type}
    </div>
    <div className="city">
      City:
      {props.city}
    </div>
    <div className="bathrooms">
      Bathrooms:
      {props.bathrooms}
    </div>
    <div className="bedrooms">
      Bedrooms:
      {props.bedrooms}
    </div>
    <div className="price">
      Price: Â£
      {props.price}
    </div>

    <button className="email-button">Email</button>

  </div>
);

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string,
  bathrooms: PropTypes.number,
  bedrooms: PropTypes.number,
  price: PropTypes.string,
  city: PropTypes.string.isRequired,
};
export default PropertyCard;
