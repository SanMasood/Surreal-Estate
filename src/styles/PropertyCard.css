.property-card{
    border: 1px solid grey;
    margin: 5px;
    padding: 0 auto;
    background-color: rgba(194, 199, 243, 0.377);
    border-radius: 5px;;
}
.property-card > *{
    padding: 5px;
    font-weight: bolder;
}
.email-button, .save-button{
    display: inline-block;
    border-radius: 10px;
    background-color: rgba(156, 156, 247, 0.479);
    border: none;    
}