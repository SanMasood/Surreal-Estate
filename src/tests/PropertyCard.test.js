import React from 'react';
import { render, getByTestId } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

test('renders property card', () => {
  const { asFragment } = render(<PropertyCard />);

  expect({ asFragment }).toMatchSnapshot();
});

test('renders title', () => {
  const { getByText } = render(<PropertyCard title="2 bedroom cottage" />);
  expect(getByText(/2 bedroom cottage/)).toHaveClass('title');
});
