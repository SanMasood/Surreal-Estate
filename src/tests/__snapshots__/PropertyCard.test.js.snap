// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders property card 1`] = `
Object {
  "asFragment": [Function],
}
`;
